splinterd/src/node/runnable/admin.rs

```diff
@@ -51,7 +53,52 @@ impl RunnableAdminSubsystem {
         let mut service_transport = self.service_transport;
         let routing_writer = self.routing_writer;
 
-        let registry = store_factory.get_registry_store();
+        let mut registry = store_factory.get_registry_store();
+
+        if let Some(external_registries) = self.registries {
+            let read_only_registries = external_registries
+                .iter()
+                .filter_map(|registry| {
```

Comment:
I’m not sure `filter_map` is the right way to go here. We should probably propagate the errors that can occur on the individual registries. 

Better to `map` and then `collect::<Result<Vec<_>, _>>()?`